openpgp-dsm/src/lib.rs

```diff
@@ -1089,33 +1089,51 @@ pub fn list_groups(cred: Credentials) -> Result<Vec<Group>> {
 /// Extracts the certificate of the corresponding PGP key. Note that this
 /// certificate, created at key-generation time, is stored in the custom
 /// metadata of the Security Object representing the primary key.
-pub fn extract_cert(key_name: &str, cred: Credentials) -> Result<Cert> {
+pub fn extract_cert(key_name: &str, cred: Credentials, key_id: Option<&str>) -> Result<Cert> {
     info!("dsm extract_cert");
     let dsm_client = cred.dsm_client()?;
 
-    let sobject = dsm_client
-        .get_sobject(None, &SobjectDescriptor::Name(key_name.to_string()))
-        .context(format!("could not get primary key {}", key_name))?;
+    // keyring will only accept key-ids
+    let kid: Option<Uuid> = key_id.and_then(|id| id.parse().ok());
+
+    let sobject_descriptor= if let Some(key_id) = kid {
+        SobjectDescriptor::Kid(key_id)
+    } else{
+        SobjectDescriptor::Name(key_name.to_string())
+    };
 
+    let sobject = dsm_client
+            .get_sobject(None, &sobject_descriptor)
+            .context(format!("could not get primary key {}", key_id.map_or_else(|| key_name.to_string(), |id| id.to_string())))?;
+    
     Cert::from_str(
         &KeyMetadata::from_sobject(&sobject)?.certificate
         .ok_or(anyhow::anyhow!("no certificate in DSM custom metadata"))?
     )
 }
 
-pub fn extract_tsk_from_dsm(key_name: &str, cred: Credentials) -> Result<Cert> {
+pub fn extract_tsk_from_dsm(key_name: &str, cred: Credentials, key_id: Option<&str>) -> Result<Cert> {
```

Comment:
instead of two arguments, where always one of them is ignored, I would introduce an enum
```
enum T {                   // Name appropriately 
    KeyName(&str),
    KeyringIds(&str),
    // other variants?
}
```
and use this as first argument for `extract_tsk_from_dsm` and others